main.py

```diff
@@ -0,0 +1,89 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+def read_root() -> dict:
+    return {"message": "Welcome to FastAPI!!!"}
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int = 0,
+    limit: int = 10,
+    db: Session = Depends(get_db)
+):
+    authors = crud.get_author_list(db=db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/authors/{author_id}", response_model=schemas.Author)
+def read_single_author(author_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author(db=db, author_id=author_id)
+
+    if db_author is None:
+        raise HTTPException(
+            status_code=404, detail="Author not found"
+        )
+    return db_author
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    existing_author = crud.get_author_by_name(db=db, name=author.name)
+    if existing_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Such name already exist"
```

Comment:
There is a minor typo in the error message: 'Such name already exist' should be 'Such name already exists'.